Bluepill-cli/Bluepill-cli/Simulator/BPSimulator.m

```diff
@@ -145,6 +169,107 @@ - (NSError *)waitForDeviceReady {
     return nil;
 }
 
++ (BOOL)installAppWithBundleID:(NSString *)hostBundleID
+                    bundlePath:(NSString *)hostBundlePath
+                        device:(SimDevice *)device
+                         error:(NSError **)error {
+
+    return [device installApplication:[NSURL fileURLWithPath:hostBundlePath]
+                          withOptions:@{kCFBundleIdentifier: hostBundleID}
+                                error:error];
+}
+
++ (BOOL)uninstallAppWithBundleID:(NSString *)hostBundleID
+                      bundlePath:(NSString *)hostBundlePath
+                          device:(SimDevice *)device
+                           error:(NSError **)error {
+    return [device uninstallApplication:hostBundleID
+                            withOptions:@{kCFBundleIdentifier: hostBundleID}
+                                  error:error];
+}
+
++ (void)createDeviceWithConfig:(BPConfiguration *)config andName:(NSString *)deviceName completion:(void (^)(NSError *, SimDevice *))completion {
+    deviceName = deviceName ?: [NSString stringWithFormat:@"BP%d", getpid()];
+    
+    // Create a new simulator with the given device/runtime
+    NSError *error;
+    SimServiceContext *sc = [SimServiceContext sharedServiceContextForDeveloperDir:config.xcodePath error:&error];
+    if (!sc) {
+        [BPUtils printInfo:ERROR withString:[NSString stringWithFormat:@"SimServiceContext failed: %@", [error localizedDescription]]];
+        return;
+    }
+    SimDeviceSet *deviceSet = [sc defaultDeviceSetWithError:&error];
+    if (!deviceSet) {
+        [BPUtils printInfo:ERROR withString:[NSString stringWithFormat:@"SimDeviceSet failed: %@", [error localizedDescription]]];
+        return;
+    }
+    [deviceSet createDeviceAsyncWithType:config.simDeviceType
+                                 runtime:config.simRuntime
+                                    name:deviceName
+                       completionHandler:^(NSError *error, SimDevice *device) {
+                           completion(error, device);
+                       }];
+    [BPUtils printInfo:INFO withString:@" ¯\\_(ツ)_/¯"];
+}
+
++ (SimDevice *)findDeviceWithConfig:(BPConfiguration *)config andDeviceID:(NSUUID *)deviceID {
```

Comment:
the deviceID could be one of config.useSimUDID or config.deleteSimUDID, so we have to keep it.